csharp/src/Ice/ConnectionI.cs

```diff
@@ -1941,6 +1941,9 @@ private bool validate(int operation)
                     throw new MarshalException($"Received ValidateConnection message with unexpected size {size}.");
                 }
                 TraceUtil.traceRecv(_readStream, this, _logger, _traceLevels);
+
+                // Client connection starts sending heartbeats once it's received the ValidateConnection message.
```

Comment:
even if it doesn't write anything (a rare situation).

On the server side, the writing of the initial ValidateConnection message schedules the first heartbeat.